src/Main Pages/Home.jsx

```diff
@@ -0,0 +1,80 @@
+import { useContext,useEffect,useState } from 'react'
+import { UserContext } from "../UserContext";
+import { Link } from "react-router-dom";
+import { getAllArticles, getTopics,getTopThreeArticlesPerTopic} from '../Api';
+import {mostPopular } from '../utils/utils'
+import { ArticlePage } from './ArticlePage';
+
+export const Home = () => {
+    const {userDetails,setUserDetails} = useContext(UserContext)
+    const [loading, setLoading] = useState(false)
+    const [allArticles, setAllArticles] = useState([])
+    const [topArticle, setTopArticle] = useState({})
+    const [topics, setTopics] = useState([])
+    const [top3perTopic, setTop3PerTopic] = useState({})
+
+    useEffect(()=>{
+        setLoading(true)
+        setAllArticles([])
+        getAllArticles().then(({articles}) => {
+            setLoading(false)
+            setAllArticles(articles)
+            setTopArticle(mostPopular(articles))
+        })
+        
+    },[])
+    useEffect(() => {
+        setTopics([])
+        getTopics().then(({topics})=>{
+            const topicSlugs = topics.map(topic => topic.slug)
+            setTopics(topicSlugs)
+        })
+    }, [allArticles])
+
+    useEffect(() => {
+        const updateTop3 = {}
+        setLoading(true)
+        for(const topic of topics){
+            getTopThreeArticlesPerTopic(topic).then(({articles}) => {
+                updateTop3[topic] = articles
+                setTop3PerTopic(updateTop3)
+                setLoading(false)
+        })
+    }},[allArticles])
+
+    if(loading) return <p>Loading...</p>
+
+    return (
+        <>
+        <h1>Home</h1>
+        <Link to = '/search' >
+        <button>See All Articles</button>
+        </Link>
+        <section className='most-popular-article'>
+            <Link to = {`/article/${topArticle.article_id}`}>
+            <img src = {topArticle.article_img_url} className='most-popular-image'></img>
+            <h2>{topArticle.title}</h2>
+            </Link>
+        </section>
+        <section className='home-top-main'>
+            {Object.keys(top3perTopic).map((topic) => {
```

Comment:
some nice logic here to have a top 3 for each topic!  But there's a lot of code in your home component.  Might be an idea to break it down into one or two components to make it clearer what is going on.